cafebabel/static/js/articles-tags.js

```diff
@@ -0,0 +1,150 @@
+class Tags {
+  constructor() {
+    const domContext = document.querySelector('.tags')
+    this.list = domContext.querySelector('.tags-list')
+    this.values = Array.from(this.list.querySelectorAll('input')).map(
+      input => input.value
+    )
+  }
+  checkSubmission(query) {
+    return query && !this._isTag(query)
+  }
+  add(query) {
+    this.values = this.values.concat(query)
+  }
+  remove(query) {
+    this.values = this.values.filter(value => value !== query)
+  }
+  display() {
+    const container = this.list.cloneNode(false)
+    this.list.replaceWith(this._createTagsList(container, this.values))
+    TagEventListener.addRemoveListener()
+  }
+  _isTag(tag) {
+    return this.values.includes(tag)
+  }
+  _createTagsList(container, tagValues) {
+    tagValues.forEach((tagValue, index) =>
+      container.insertAdjacentHTML(
+        'beforeend',
+        this._createTag(tagValue, index)
+      )
+    )
+
+    return container
+  }
+  _createTag(tagValue, index) {
+    return `
+      <li>
+        ${tagValue}
```

Comment:
Bad news, with that implementation [extra space bug](https://github.com/cafebabel/cafebabel.com/issues/151) stroked again, this time on the JS side 🙃

Reply:
We wrote it as one only line, and it fixes it.

Reply:
Where/when? The current implementation still adds it.